types/ember__owner/index.d.ts

```diff
@@ -0,0 +1,110 @@
+// Type definitions for non-npm package @ember/owner 4.0
+// Project: https://emberjs.com/api/ember/4.0/
+// Definitions by: Chris Krycho <https://github.com/chriskrycho>
+//                 Dan Freeman <https://github.com/dfreeman>
+//                 James C. Davis <https://github.com/jamescdavis>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.4
+
+/**
+ * The name for a factory consists of a namespace and the name of a specific
+ * type within that namespace, like `'service:session'`.
+ */
+type FullName = `${string}:${string}`;
```

Comment:
I wonder if we're going to be causing pain by not exporting this type?

Reply:
I don’t *think* so—I had the same concern, but in the tests, it does what you’d expect from a type alias and just shows the type it’s an alias for.

<img width="548" alt="Screen Shot 2022-08-02 at 08 02 44" src="https://user-images.githubusercontent.com/2403023/182393857-3edca308-e3c1-4428-a808-23e1e41602f9.png">

Reply:
@dfreeman and I ended up deciding this *should* be exported and [updated the RFC with that and a few other changes](https://github.com/emberjs/rfcs/pull/834) which came out of this review. 👍🏼